code/__DEFINES/components.dm

```diff
@@ -280,6 +280,7 @@
 #define COMSIG_ITEM_MARK_RETRIEVAL "item_mark_retrieval"			//called before marking an object for retrieval, checked in /obj/effect/proc_holder/spell/targeted/summonitem/cast() : (mob/user)
 #define COMPONENT_BLOCK_MARK_RETRIEVAL 1
 #define COMSIG_ITEM_HIT_REACT "item_hit_react"					//from base of obj/item/hit_reaction(): (list/args)
+#define COMSIG_ITEM_REFUND	"item_refund"						//for tc refunding items: (mob/living/user); returns FALSE if okay to refund, TRUE if not
```

Comment:
feels kinda weird to return TRUE on something NOT okay to refund, and vice-versa

Reply:
it's because the results of all signals are OR'd together. 1 & 1 = 1, 0 & 1 = 1, 0 & 0  = 0

Reply:
so, something saying "no refund" will take priority above "no refund"

Reply:
Find a solution then.

Reply:
Alright, done.